### ISSUE TITLE
Unable to load webUI when VPN enabled after latest ubuntu server update

### ISSUE THREAD

**Issue by mchamp00** [binhex/arch-qbittorrentvpn#294]
Ubuntu server 22.04.5 LTS
docker-compose
binhex/arch-qbittorrentvpn:latest
ExpressVPN

After doing an apt update/upgrade today, I haven't been able to access the webUI of my qbittorrent. I also pulled new docker images, but the qbittorrent image was already the latest version.

When I disable the VPN in the docker-compose entry for the container, the webUI loads up just fine. When I reenable the VPN, it stops working again. Replaced my config.ovpn with a fresh version from the ExpressVPN website, and checked my credentials in the credentials.conf file. 

supervisord.log shows the following error:
```
2024-12-09 19:55:32,666 DEBG 'start-script' stdout output:
2024-12-09 19:55:32 PUSH: Received control message: 'PUSH_REPLY,topology net30,route 10.39.0.1,redirect-gateway def1,dhcp-option DNS 10.39.0.1,comp-lzo no,ping 10,ping-restart 60,ifconfig 10.39.132.241 10.39.132.242,peer-id 131,cipher AES-256-GCM,protocol-flags cc-exit tls-ekm dyn-tls-crypt,tun-mtu 1500'
2024-12-09 19:55:32 OPTIONS IMPORT: --ifconfig/up options modified
2024-12-09 19:55:32 OPTIONS IMPORT: route options modified
2024-12-09 19:55:32 OPTIONS IMPORT: --ip-win32 and/or --dhcp-option options modified
2024-12-09 19:55:32 OPTIONS IMPORT: tun-mtu set to 1500
2024-12-09 19:55:32 net_route_v4_best_gw query: dst 0.0.0.0
2024-12-09 19:55:32 net_route_v4_best_gw result: via 172.18.0.1 dev eth0
2024-12-09 19:55:32 ROUTE_GATEWAY 172.18.0.1/255.255.0.0 IFACE=eth0 HWADDR=02:42:ac:12:00:16
2024-12-09 19:55:32 ERROR: Cannot open TUN/TAP dev /dev/net/tun: Operation not permitted (errno=1)
2024-12-09 19:55:32 Exiting due to fatal error
```

It looks like something must have changed with /dev/net/tun during the last update? I have no idea what this file is/does. I tried changing the chmod of /dev/net/tun to 777 in case it was a permissions issue, but that didn't fix the problem.

**Comment by Arron17** [comment]
Same problem for me

Fedora 6.11.8-200.fc40.x86_64
docker-compose
binhex/arch-qbittorrentvpn:latest
Mullvad VPN

from supervisord.log, this just loops over and over:

```
2024-12-10 05:13:07,942 DEBG 'start-script' stdout output:  
2024-12-10 05:13:07 Pushed option removed by filter: 'ifconfig-ipv6 fdda:d0d0:cafe:1196::101a/64 fdda:d0d0:cafe:1196::'  
2024-12-10 05:13:07 OPTIONS IMPORT: --socket-flags option modified  
2024-12-10 05:13:07 NOTE: setsockopt TCP_NODELAY=1 failed  
2024-12-10 05:13:07 OPTIONS IMPORT: --ifconfig/up options modified  
2024-12-10 05:13:07 OPTIONS IMPORT: route options modified  
2024-12-10 05:13:07 OPTIONS IMPORT: route-related options modified  
2024-12-10 05:13:07 OPTIONS IMPORT: --ip-win32 and/or --dhcp-option options modified  
2024-12-10 05:13:07 OPTIONS IMPORT: tun-mtu set to 1500  
2024-12-10 05:13:07 net_route_v4_best_gw query: dst 0.0.0.0                                                                                                                                                                                                                                                                                                     2024-12-10 05:13:07,942 DEBG 'start-script' stdout output:  
2024-12-10 05:13:07 net_route_v4_best_gw result: via 172.19.0.1 dev eth0  
2024-12-10 05:13:07 ROUTE_GATEWAY 172.19.0.1/255.255.0.0 IFACE=eth0 HWADDR=02:42:ac:13:00:04                                                                                                                                                                                                                                                           2024-12-10 05:13:07,942 DEBG 'start-script' stdout output:  
2024-12-10 05:13:07 ERROR: Cannot open TUN/TAP dev /dev/net/tun: Operation not permitted (errno=1)  
2024-12-10 05:13:07 Exiting due to fatal error
```

If it's helpful these are all the components that were updated to cause it to break

```
Install  cups-filesystem-1:2.4.11-8.fc40.noarch                @updates
    Upgrade  containerd.io-1.7.24-3.1.fc40.x86_64                  @docker-ce-stable
    Upgraded containerd.io-1.7.23-3.1.fc40.x86_64                  @@System
    Upgrade  binutils-2.41-38.fc40.x86_64                          @updates
    Upgraded binutils-2.41-37.fc40.x86_64                          @@System
    Upgrade  binutils-gold-2.41-38.fc40.x86_64                     @updates
    Upgraded binutils-gold-2.41-37.fc40.x86_64                     @@System
    Upgrade  btrfs-progs-6.12-1.fc40.x86_64                        @updates
    Upgraded btrfs-progs-6.11-1.fc40.x86_64                        @@System
    Upgrade  cockpit-329.1-1.fc40.x86_64                           @updates
    Upgraded cockpit-329-1.fc40.x86_64                             @@System
    Upgrade  cockpit-bridge-329.1-1.fc40.x86_64                    @updates
    Upgraded cockpit-bridge-329-1.fc40.x86_64                      @@System
    Upgrade  cockpit-networkmanager-329.1-1.fc40.noarch            @updates
    Upgraded cockpit-networkmanager-329-1.fc40.noarch              @@System
    Upgrade  cockpit-packagekit-329.1-1.fc40.noarch                @updates
    Upgraded cockpit-packagekit-329-1.fc40.noarch                  @@System
    Upgrade  cockpit-selinux-329.1-1.fc40.noarch                   @updates
    Upgraded cockpit-selinux-329-1.fc40.noarch                     @@System
    Upgrade  cockpit-storaged-329.1-1.fc40.noarch                  @updates
    Upgraded cockpit-storaged-329-1.fc40.noarch                    @@System
    Upgrade  cockpit-system-329.1-1.fc40.noarch                    @updates
    Upgraded cockpit-system-329-1.fc40.noarch                      @@System
    Upgrade  cockpit-ws-329.1-1.fc40.x86_64                        @updates
    Upgraded cockpit-ws-329-1.fc40.x86_64                          @@System
    Upgrade  container-selinux-2:2.234.2-1.fc40.noarch             @updates
    Upgraded container-selinux-2:2.233.0-1.fc40.noarch             @@System
    Upgrade  cups-libs-1:2.4.11-8.fc40.x86_64                      @updates
    Upgraded cups-libs-1:2.4.11-1.fc40.x86_64                      @@System
    Upgrade  dnf-4.22.0-1.fc40.noarch                              @updates
    Upgraded dnf-4.21.1-1.fc40.noarch                              @@System
    Upgrade  dnf-automatic-4.22.0-1.fc40.noarch                    @updates
    Upgraded dnf-automatic-4.21.1-1.fc40.noarch                    @@System
    Upgrade  dnf-data-4.22.0-1.fc40.noarch                         @updates
    Upgraded dnf-data-4.21.1-1.fc40.noarch                         @@System
    Upgrade  dnf-plugins-core-4.10.0-1.fc40.noarch                 @updates
    Upgraded dnf-plugins-core-4.9.0-1.fc40.noarch                  @@System
    Upgrade  elfutils-0.192-7.fc40.x86_64                          @updates
    Upgraded elfutils-0.192-6.fc40.x86_64                          @@System
    Upgrade  elfutils-debuginfod-client-0.192-7.fc40.x86_64        @updates
    Upgraded elfutils-debuginfod-client-0.192-6.fc40.x86_64        @@System
    Upgrade  elfutils-default-yama-scope-0.192-7.fc40.noarch       @updates
    Upgraded elfutils-default-yama-scope-0.192-6.fc40.noarch       @@System
    Upgrade  elfutils-libelf-0.192-7.fc40.x86_64                   @updates
    Upgraded elfutils-libelf-0.192-6.fc40.x86_64                   @@System
    Upgrade  elfutils-libelf-devel-0.192-7.fc40.x86_64             @updates
    Upgraded elfutils-libelf-devel-0.192-6.fc40.x86_64             @@System
    Upgrade  elfutils-libs-0.192-7.fc40.x86_64                     @updates
    Upgraded elfutils-libs-0.192-6.fc40.x86_64                     @@System
    Upgrade  exfatprogs-1.2.6-1.fc40.x86_64                        @updates
    Upgraded exfatprogs-1.2.5-1.fc40.x86_64                        @@System
    Upgrade  fprintd-1.94.4-1.fc40.x86_64                          @updates
    Upgraded fprintd-1.94.2-11.fc40.x86_64                         @@System
    Upgrade  fprintd-pam-1.94.4-1.fc40.x86_64                      @updates
    Upgraded fprintd-pam-1.94.2-11.fc40.x86_64                     @@System
    Upgrade  git-2.47.1-1.fc40.x86_64                              @updates
    Upgraded git-2.47.0-1.fc40.x86_64                              @@System
    Upgrade  git-core-2.47.1-1.fc40.x86_64                         @updates
    Upgraded git-core-2.47.0-1.fc40.x86_64                         @@System
    Upgrade  git-core-doc-2.47.1-1.fc40.noarch                     @updates
    Upgraded git-core-doc-2.47.0-1.fc40.noarch                     @@System
    Upgrade  hwdata-0.390-1.fc40.noarch                            @updates
    Upgraded hwdata-0.389-1.fc40.noarch                            @@System
    Upgrade  kexec-tools-2.0.30-1.fc40.x86_64                      @updates
    Upgraded kexec-tools-2.0.29-1.fc40.x86_64                      @@System
    Upgrade  libblockdev-3.1.1-2.fc40.x86_64                       @updates
    Upgraded libblockdev-3.1.1-1.fc40.x86_64                       @@System
    Upgrade  libblockdev-btrfs-3.1.1-2.fc40.x86_64                 @updates
    Upgraded libblockdev-btrfs-3.1.1-1.fc40.x86_64                 @@System
    Upgrade  libblockdev-crypto-3.1.1-2.fc40.x86_64                @updates
    Upgraded libblockdev-crypto-3.1.1-1.fc40.x86_64                @@System
    Upgrade  libblockdev-fs-3.1.1-2.fc40.x86_64                    @updates
    Upgraded libblockdev-fs-3.1.1-1.fc40.x86_64                    @@System
    Upgrade  libblockdev-loop-3.1.1-2.fc40.x86_64                  @updates
    Upgraded libblockdev-loop-3.1.1-1.fc40.x86_64                  @@System
    Upgrade  libblockdev-lvm-3.1.1-2.fc40.x86_64                   @updates
    Upgraded libblockdev-lvm-3.1.1-1.fc40.x86_64                   @@System
    Upgrade  libblockdev-mdraid-3.1.1-2.fc40.x86_64                @updates
    Upgraded libblockdev-mdraid-3.1.1-1.fc40.x86_64                @@System
    Upgrade  libblockdev-nvme-3.1.1-2.fc40.x86_64                  @updates
    Upgraded libblockdev-nvme-3.1.1-1.fc40.x86_64                  @@System
    Upgrade  libblockdev-part-3.1.1-2.fc40.x86_64                  @updates
    Upgraded libblockdev-part-3.1.1-1.fc40.x86_64                  @@System
    Upgrade  libblockdev-swap-3.1.1-2.fc40.x86_64                  @updates
    Upgraded libblockdev-swap-3.1.1-1.fc40.x86_64                  @@System
    Upgrade  libblockdev-utils-3.1.1-2.fc40.x86_64                 @updates
    Upgraded libblockdev-utils-3.1.1-1.fc40.x86_64                 @@System
    Upgrade  libdnf-0.73.4-1.fc40.x86_64                           @updates
    Upgraded libdnf-0.73.3-1.fc40.x86_64                           @@System
    Upgrade  libldb-2.9.2-2.fc40.x86_64                            @updates
    Upgraded libldb-2.9.1-4.fc40.x86_64                            @@System
    Upgrade  libnetapi-2:4.20.6-1.fc40.x86_64                      @updates
    Upgraded libnetapi-2:4.20.5-1.fc40.x86_64                      @@System
    Upgrade  libnfsidmap-1:2.8.1-2.rc2.fc40.x86_64                 @updates
    Upgraded libnfsidmap-1:2.8.1-1.rc1.fc40.x86_64                 @@System
    Upgrade  libsmbclient-2:4.20.6-1.fc40.x86_64                   @updates
    Upgraded libsmbclient-2:4.20.5-1.fc40.x86_64                   @@System
    Upgrade  libsolv-0.7.31-1.fc40.x86_64                          @updates
    Upgraded libsolv-0.7.30-1.fc40.x86_64                          @@System
    Upgrade  libsoup3-3.4.4-5.fc40.x86_64                          @updates
    Upgraded libsoup3-3.4.4-3.fc40.x86_64                          @@System
    Upgrade  libwbclient-2:4.20.6-1.fc40.x86_64                    @updates
    Upgraded libwbclient-2:4.20.5-1.fc40.x86_64                    @@System
    Upgrade  libxcrypt-4.4.36-11.fc40.x86_64                       @updates
    Upgraded libxcrypt-4.4.36-10.fc40.x86_64                       @@System
    Upgrade  libxcrypt-compat-4.4.36-11.fc40.x86_64                @updates
    Upgraded libxcrypt-compat-4.4.36-10.fc40.x86_64                @@System
    Upgrade  libxcrypt-devel-4.4.36-11.fc40.x86_64                 @updates
    Upgraded libxcrypt-devel-4.4.36-10.fc40.x86_64                 @@System
    Upgrade  mtools-4.0.46-1.fc40.x86_64                           @updates
    Upgraded mtools-4.0.45-1.fc40.x86_64                           @@System
    Upgrade  nfs-utils-1:2.8.1-2.rc2.fc40.x86_64                   @updates
    Upgraded nfs-utils-1:2.8.1-1.rc1.fc40.x86_64                   @@System
    Upgrade  nspr-4.36.0-1.fc40.x86_64                             @updates
    Upgraded nspr-4.35.0-30.fc40.x86_64                            @@System
    Upgrade  nss-3.106.0-1.fc40.x86_64                             @updates
    Upgraded nss-3.105.0-1.fc40.x86_64                             @@System
    Upgrade  nss-softokn-3.106.0-1.fc40.x86_64                     @updates
    Upgraded nss-softokn-3.105.0-1.fc40.x86_64                     @@System
    Upgrade  nss-softokn-freebl-3.106.0-1.fc40.x86_64              @updates
    Upgraded nss-softokn-freebl-3.105.0-1.fc40.x86_64              @@System
    Upgrade  nss-sysinit-3.106.0-1.fc40.x86_64                     @updates
    Upgraded nss-sysinit-3.105.0-1.fc40.x86_64                     @@System
    Upgrade  nss-util-3.106.0-1.fc40.x86_64                        @updates
    Upgraded nss-util-3.105.0-1.fc40.x86_64                        @@System
    Upgrade  pam-1.6.1-5.fc40.x86_64                               @updates
    Upgraded pam-1.6.1-4.fc40.x86_64                               @@System
    Upgrade  pam-libs-1.6.1-5.fc40.x86_64                          @updates
    Upgraded pam-libs-1.6.1-4.fc40.x86_64                          @@System
    Upgrade  perl-Git-2.47.1-1.fc40.noarch                         @updates
    Upgraded perl-Git-2.47.0-1.fc40.noarch                         @@System
    Upgrade  perl-Module-CoreList-1:5.20241120-1.fc40.noarch       @updates
    Upgraded perl-Module-CoreList-1:5.20241020-1.fc40.noarch       @@System
    Upgrade  perl-Module-CoreList-tools-1:5.20241120-1.fc40.noarch @updates
    Upgraded perl-Module-CoreList-tools-1:5.20241020-1.fc40.noarch @@System
    Upgrade  pyproject-srpm-macros-1.16.3-1.fc40.noarch            @updates
    Upgraded pyproject-srpm-macros-1.16.0-1.fc40.noarch            @@System
    Upgrade  python3-argcomplete-3.5.2-1.fc40.noarch               @updates
    Upgraded python3-argcomplete-3.5.1-1.fc40.noarch               @@System
    Upgrade  python3-boto3-1.35.76-1.fc40.noarch                   @updates
    Upgraded python3-boto3-1.35.63-1.fc40.noarch                   @@System
    Upgrade  python3-botocore-1.35.76-1.fc40.noarch                @updates
    Upgraded python3-botocore-1.35.63-1.fc40.noarch                @@System
    Upgrade  python3-dnf-4.22.0-1.fc40.noarch                      @updates
    Upgraded python3-dnf-4.21.1-1.fc40.noarch                      @@System
    Upgrade  python3-dnf-plugins-core-4.10.0-1.fc40.noarch         @updates
    Upgraded python3-dnf-plugins-core-4.9.0-1.fc40.noarch          @@System
    Upgrade  python3-hawkey-0.73.4-1.fc40.x86_64                   @updates
    Upgraded python3-hawkey-0.73.3-1.fc40.x86_64                   @@System
    Upgrade  python3-ldb-2.9.2-2.fc40.x86_64                       @updates
    Upgraded python3-ldb-2.9.1-4.fc40.x86_64                       @@System
    Upgrade  python3-libdnf-0.73.4-1.fc40.x86_64                   @updates
    Upgraded python3-libdnf-0.73.3-1.fc40.x86_64                   @@System
    Upgrade  python3-s3transfer-0.10.4-1.fc40.noarch               @updates
    Upgraded python3-s3transfer-0.10.3-1.fc40.noarch               @@System
    Upgrade  python3-samba-2:4.20.6-1.fc40.x86_64                  @updates
    Upgraded python3-samba-2:4.20.5-1.fc40.x86_64                  @@System
    Upgrade  python3-samba-dc-2:4.20.6-1.fc40.x86_64               @updates
    Upgraded python3-samba-dc-2:4.20.5-1.fc40.x86_64               @@System
    Upgrade  rpm-sequoia-1.7.0-3.fc40.x86_64                       @updates
    Upgraded rpm-sequoia-1.7.0-1.fc40.x86_64                       @@System
    Upgrade  samba-2:4.20.6-1.fc40.x86_64                          @updates
    Upgraded samba-2:4.20.5-1.fc40.x86_64                          @@System
    Upgrade  samba-client-2:4.20.6-1.fc40.x86_64                   @updates
    Upgraded samba-client-2:4.20.5-1.fc40.x86_64                   @@System
    Upgrade  samba-client-libs-2:4.20.6-1.fc40.x86_64              @updates
    Upgraded samba-client-libs-2:4.20.5-1.fc40.x86_64              @@System
    Upgrade  samba-common-2:4.20.6-1.fc40.noarch                   @updates
    Upgraded samba-common-2:4.20.5-1.fc40.noarch                   @@System
    Upgrade  samba-common-libs-2:4.20.6-1.fc40.x86_64              @updates
    Upgraded samba-common-libs-2:4.20.5-1.fc40.x86_64              @@System
    Upgrade  samba-common-tools-2:4.20.6-1.fc40.x86_64             @updates
    Upgraded samba-common-tools-2:4.20.5-1.fc40.x86_64             @@System
    Upgrade  samba-dc-libs-2:4.20.6-1.fc40.x86_64                  @updates
    Upgraded samba-dc-libs-2:4.20.5-1.fc40.x86_64                  @@System
    Upgrade  samba-dcerpc-2:4.20.6-1.fc40.x86_64                   @updates
    Upgraded samba-dcerpc-2:4.20.5-1.fc40.x86_64                   @@System
    Upgrade  samba-ldb-ldap-modules-2:4.20.6-1.fc40.x86_64         @updates
    Upgraded samba-ldb-ldap-modules-2:4.20.5-1.fc40.x86_64         @@System
    Upgrade  samba-libs-2:4.20.6-1.fc40.x86_64                     @updates
    Upgraded samba-libs-2:4.20.5-1.fc40.x86_64                     @@System
    Upgrade  setroubleshoot-server-3.3.35-1.fc40.x86_64            @updates
    Upgraded setroubleshoot-server-3.3.34-1.fc40.x86_64            @@System
    Upgrade  systemd-255.15-1.fc40.x86_64                          @updates
    Upgraded systemd-255.14-1.fc40.x86_64                          @@System
    Upgrade  systemd-boot-unsigned-255.15-1.fc40.x86_64            @updates
    Upgraded systemd-boot-unsigned-255.14-1.fc40.x86_64            @@System
    Upgrade  systemd-libs-255.15-1.fc40.x86_64                     @updates
    Upgraded systemd-libs-255.14-1.fc40.x86_64                     @@System
    Upgrade  systemd-networkd-255.15-1.fc40.x86_64                 @updates
    Upgraded systemd-networkd-255.14-1.fc40.x86_64                 @@System
    Upgrade  systemd-oomd-defaults-255.15-1.fc40.noarch            @updates
    Upgraded systemd-oomd-defaults-255.14-1.fc40.noarch            @@System
    Upgrade  systemd-pam-255.15-1.fc40.x86_64                      @updates
    Upgraded systemd-pam-255.14-1.fc40.x86_64                      @@System
    Upgrade  systemd-resolved-255.15-1.fc40.x86_64                 @updates
    Upgraded systemd-resolved-255.14-1.fc40.x86_64                 @@System
    Upgrade  systemd-udev-255.15-1.fc40.x86_64                     @updates
    Upgraded systemd-udev-255.14-1.fc40.x86_64                     @@System
    Upgrade  tcl-1:8.6.13-3.fc40.x86_64                            @updates
    Upgraded tcl-1:8.6.13-2.fc40.x86_64                            @@System
    Upgrade  vim-data-2:9.1.906-1.fc40.noarch                      @updates
    Upgraded vim-data-2:9.1.866-1.fc40.noarch                      @@System
    Upgrade  vim-minimal-2:9.1.906-1.fc40.x86_64                   @updates
    Upgraded vim-minimal-2:9.1.866-1.fc40.x86_64                   @@System
    Upgrade  wget2-2.2.0-1.fc40.x86_64                             @updates
    Upgraded wget2-2.1.0-11.fc40.x86_64                            @@System
    Upgrade  wget2-libs-2.2.0-1.fc40.x86_64                        @updates
    Upgraded wget2-libs-2.1.0-11.fc40.x86_64                       @@System
    Upgrade  wget2-wget-2.2.0-1.fc40.x86_64                        @updates
    Upgraded wget2-wget-2.1.0-11.fc40.x86_64                       @@System
    Upgrade  yum-4.22.0-1.fc40.noarch                              @updates
    Upgraded yum-4.21.1-1.fc40.noarch                              @@System        
```

**Comment by binhex** [comment]
> Ubuntu server 22.04.5 LTS docker-compose binhex/arch-qbittorrentvpn:latest ExpressVPN
> 
> After doing an apt update/upgrade today, I haven't been able to access the webUI of my qbittorrent. I also pulled new docker images, but the qbittorrent image was already the latest version.
> 
> When I disable the VPN in the docker-compose entry for the container, the webUI loads up just fine. When I reenable the VPN, it stops working again. Replaced my config.ovpn with a fresh version from the ExpressVPN website, and checked my credentials in the credentials.conf file.
> 
> supervisord.log shows the following error:
> 
> ```
> 2024-12-09 19:55:32,666 DEBG 'start-script' stdout output:
> 2024-12-09 19:55:32 PUSH: Received control message: 'PUSH_REPLY,topology net30,route 10.39.0.1,redirect-gateway def1,dhcp-option DNS 10.39.0.1,comp-lzo no,ping 10,ping-restart 60,ifconfig 10.39.132.241 10.39.132.242,peer-id 131,cipher AES-256-GCM,protocol-flags cc-exit tls-ekm dyn-tls-crypt,tun-mtu 1500'
> 2024-12-09 19:55:32 OPTIONS IMPORT: --ifconfig/up options modified
> 2024-12-09 19:55:32 OPTIONS IMPORT: route options modified
> 2024-12-09 19:55:32 OPTIONS IMPORT: --ip-win32 and/or --dhcp-option options modified
> 2024-12-09 19:55:32 OPTIONS IMPORT: tun-mtu set to 1500
> 2024-12-09 19:55:32 net_route_v4_best_gw query: dst 0.0.0.0
> 2024-12-09 19:55:32 net_route_v4_best_gw result: via 172.18.0.1 dev eth0
> 2024-12-09 19:55:32 ROUTE_GATEWAY 172.18.0.1/255.255.0.0 IFACE=eth0 HWADDR=02:42:ac:12:00:16
> 2024-12-09 19:55:32 ERROR: Cannot open TUN/TAP dev /dev/net/tun: Operation not permitted (errno=1)
> 2024-12-09 19:55:32 Exiting due to fatal error
> ```
> 
> It looks like something must have changed with /dev/net/tun during the last update? I have no idea what this file is/does. I tried changing the chmod of /dev/net/tun to 777 in case it was a permissions issue, but that didn't fix the problem.

what is your docker run command/compose file look like?

**Comment by Arron17** [comment]
It does look like it's some Linux component that has broken it. I've rolled back to the previous versions of that update list I posted and it's working again. Unfortunately I don't know enough about Linux to start guessing which update caused it.

**Comment by mchamp00** [comment]
> It looks to be containerd.io that has broken it. I went through my updates removing them 1 by 1 until it started working and then replaced all but one that I thought could cause the issue.
> 
> Downgrading from "containerd.io-1.7.24-3.1.fc40.x86_64" to "containerd.io-1.7.23-3.1.fc40.x86_64" made it work again.
> 
> @mchamp00 Can you try downgrading your version of containerd.io and see if it starts working again?

I decided to upgrade everything to Ubuntu server 24.04.1 LTS and start from scratch since it's been a while since I cleaned things up on the server. If the same problem occurs after the upgrade, I'll try your recommendation and let you know.

**Comment by binhex** [comment]
hi guys i think [this ](https://github.com/binhex/arch-delugevpn/issues/418#issuecomment-2525061517) might be related to your issue

**Comment by binhex** [comment]
> So not sure if this is going to gradually affect other people as they upgrade

Yep this is my concern, i guess i will need to change the docs to reflect this change.

**Comment by nillestimothy** [comment]
> > hi guys i think [this ](https://github.com/binhex/arch-delugevpn/issues/418#issuecomment-2525061517) might be related to your issue.
> > Also check out [this ](https://github.com/containerd/containerd/issues/11078) linked issue, looks like a pretty close match to me.
> 
> I've updated to the latest containerd again and added this to the docker-compose file for the container
> 
> ```
> devices:
>       - /dev/net/tun:/dev/net/tun
> ```
> 
> That has seemed to have fixed it. From those links it sounds like access to that device should never have been enabled by default in the first place, so they've disabled it. So not sure if this is going to gradually affect other people as they upgrade.

Confirming this has worked for both Ubuntu 22.04 and Ubuntu 24.04

**Comment by mchamp00** [comment]
> > hi guys i think [this ](https://github.com/binhex/arch-delugevpn/issues/418#issuecomment-2525061517) might be related to your issue.
> > Also check out [this ](https://github.com/containerd/containerd/issues/11078) linked issue, looks like a pretty close match to me.
> 
> I've updated to the latest containerd again and added this to the docker-compose file for the container
> 
> ```
> devices:
>       - /dev/net/tun:/dev/net/tun
> ```
> 
> That has seemed to have fixed it. From those links it sounds like access to that device should never have been enabled by default in the first place, so they've disabled it. So not sure if this is going to gradually affect other people as they upgrade.

This worked for me as well. Thanks!

**Comment by Anthony-Lloyd** [comment]
> > hi guys i think [this ](https://github.com/binhex/arch-delugevpn/issues/418#issuecomment-2525061517) might be related to your issue.
> > Also check out [this ](https://github.com/containerd/containerd/issues/11078) linked issue, looks like a pretty close match to me.
> 
> I've updated to the latest containerd again and added this to the docker-compose file for the container
> 
> ```
> devices:
>       - /dev/net/tun:/dev/net/tun
> ```
> 
> That has seemed to have fixed it. From those links it sounds like access to that device should never have been enabled by default in the first place, so they've disabled it. So not sure if this is going to gradually affect other people as they upgrade.

confirmed to work for me on debian 12.
just ran apt full-upgrade and this container broke but easily resolved

**Comment by nillestimothy** [comment]
So this is all due to ContainerD updating its version of runc, which removed default access to TUN/TAP - which is why we have to statically set the containers access to the host device.

Now if anyone has a solution to make this work for swarm users with the updated version of ContainerD, I would love your recommendations 😬

**Comment by tessierp** [comment]
> So this is all due to ContainerD updating its version of runc, which removed default access to TUN/TAP - which is why we have to statically set the containers access to the host device.
> 
> Now if anyone has a solution to make this work for swarm users with the updated version of ContainerD, I would love your recommendations 😬

I am in the same boat as you. Would appreciate a solution / fix.

**Comment by nillestimothy** [comment]
> I am in the same boat as you. Would appreciate a solution / fix.

Best way forward for now is to hardpin the prior containerd version.

Runc did revert the change but it hasn't released yet and containerd will need to update the newer runc version

**Comment by tessierp** [comment]
> > I am in the same boat as you. Would appreciate a solution / fix.
> 
> Best way forward for now is to hardpin the prior containerd version.
> 
> Runc did revert the change but it hasn't released yet and containerd will need to update the newer runc version

Just a quick question, how would I hardpin the containerd version?

If they will revert the changes and release then I guess I don't have to worry about too much, shouldn't be that long.

**Comment by nillestimothy** [comment]
> Just a quick question, how would I hardpin the containerd version?
> 
> 
> 
> If they will revert the changes and release then I guess I don't have to worry about too much, shouldn't be that long.

It might be a few months I am sure until it finds its way through both runc and containerd

To hardpin containerd:

sudo apt install containerd.io=1.7.23-1

sudo apt-mark hold containerd

**Comment by tessierp** [comment]
> > Just a quick question, how would I hardpin the containerd version?
> > If they will revert the changes and release then I guess I don't have to worry about too much, shouldn't be that long.
> 
> It might be a few months I am sure until it finds its way through both runc and containerd
> 
> To hardpin containerd:
> 
> sudo apt install containerd.io=1.7.23-1
> 
> sudo apt-mark hold containerd

That would work even after the upgrade? And I presume I have to do this for all my nodes.

Another question, once marked, how to do I undo the mark?

**Comment by mchamp00** [comment]
> Another question, once marked, how to do I undo the mark?

sudo apt-mark unhold containerd

**Comment by tessierp** [comment]
> > Another question, once marked, how to do I undo the mark?
> 
> sudo apt-mark unhold containerd

Easy stuff. Thanks for the help!

**Comment by tessierp** [comment]
That didn't seem to work for me.. qbittorrent with vpn is still borked.. I restarted the service of course. Perhaps I need to reboot?

**Comment by tessierp** [comment]
Nope.. no joy... when VPN is on still no access to the WEBAPI with containerd 1.7.23-1

**Comment by Biggingus** [comment]
any idea how to implement this fix for Unraid OS 
Thanks

**Comment by tessierp** [comment]
Was that last merge supposed to fix anything related to this issue?

**Comment by nillestimothy** [comment]
> Was that last merge supposed to fix anything related to this issue? 

Are you able to reference the merge you are speaking about?

**Comment by tessierp** [comment]
Referring to this, I assumed that by referencing this issue something was done maybe fix the issue.

![image](https://github.com/user-attachments/assets/4ab9f64e-9c9b-4f26-8ad7-1b41df11bf61)

**Comment by nillestimothy** [comment]
> Referring to this, I assumed that by referencing this issue something was done maybe fix the issue.
> 
> 
> 
> ![image](https://github.com/user-attachments/assets/4ab9f64e-9c9b-4f26-8ad7-1b41df11bf61)
> 
> 

No this was someone else's personnel repo that they referenced this issue. 

The issue is a two fold issue. This won't be fixed until RUNC release version 1.2.4 and then containerD will need to update their end to use that newer version of RUNC.
